### PR TITLE
fix config generation on multiple threads

### LINKED ISSUE #3193: Using SunPy in two threads sometimes crashes when creating config
I test HelioPy on two threads at the same time, and when it comes to importing SunPy the following error happens sporadically:

```
heliopy\coordinates\test\test_transformations.py:1: in <module>
    import heliopy.coordinates.frames as frames
heliopy\coordinates\frames.py:19: in <module>
    import sunpy.coordinates.sun
c:\hostedtoolcache\windows\python\3.6.8\x64\lib\site-packages\sunpy\__init__.py:47: in <module>
    config = load_config()
c:\hostedtoolcache\windows\python\3.6.8\x64\lib\site-packages\sunpy\util\config.py:30: in load_config
    config_files = _find_config_files()
c:\hostedtoolcache\windows\python\3.6.8\x64\lib\site-packages\sunpy\util\config.py:69: in _find_config_files
    config_path = Path(_get_user_configdir())
c:\hostedtoolcache\windows\python\3.6.8\x64\lib\site-packages\sunpy\util\config.py:137: in _get_user_configdir
    if not _is_writable_dir(configdir):
c:\hostedtoolcache\windows\python\3.6.8\x64\lib\site-packages\sunpy\util\config.py:123: in _is_writable_dir
    Path(p).mkdir(parents=True)
c:\hostedtoolcache\windows\python\3.6.8\x64\lib\pathlib.py:1253: in mkdir
    self.mkdir(mode, parents=False, exist_ok=exist_ok)
c:\hostedtoolcache\windows\python\3.6.8\x64\lib\pathlib.py:1248: in mkdir
    self._accessor.mkdir(self, mode)
c:\hostedtoolcache\windows\python\3.6.8\x64\lib\pathlib.py:387: in wrapped
    return strfunc(str(pathobj), *args)
E   FileExistsError: [WinError 183] Cannot create a file when that file already exists: 'C:\\Users\\VssAdministrator\\AppData\\Local\\sunpy\\sunpy'
```

I think the solution is to change `Path(p).mkdir(parents=True)` to `Path(p).mkdir(parents=True, exist_okay=True)`

### PR DESCRIPTION
**Pull request by nabobalis**
Fixes #3193 (I don't have a test for this tho.)

### CODE REVIEW

**Comment by dstansby** [comment]
Thanks!

**Comment by ayshih** [comment]
Given the concern parallel threads/processes, it's better to get rid of the `if..then` structure.  You're also not checking the edge case where a file might get created between the `exists()` call and the `mkdir()` call, which will raise a `FileExistsError` exception.

```python
    if not Path(p).exists():
        Path(p).mkdir(parents=True, exist_ok=True)
    return Path(p).is_dir() and os.access(p, os.W_OK)
```

could be changed to

```python
    try:
        Path(p).mkdir(parents=True, exist_ok=True)
    except FileExistsError:
        return false
    else:
        return os.access(p, os.W_OK)
```

**Comment by nabobalis** [comment]
Would a hybrid work? 

```
    if not Path(p).exists():
        try:
            Path(p).mkdir(parents=True)
        except FileExistsError:
            return Path(p).is_dir() and os.access(p, os.W_OK)
    else:
        return os.access(p, os.W_OK)
```

**Comment by ayshih** [comment]
My code block takes advantage of the fact that `mkdir()` with `parents=True` and `exist_ok=True` only has two outcomes: either the directory (now) exists, or a `FileExistsError` is raised because there is a file there instead of a directory.  You don't need all of the extra checks.

Your latest code block has a bug in the top-level `if..then` because it doesn't check whether the path that exists is a directory.  Plus, it's much longer than my version.

**Comment by nabobalis** [comment]
I changed the fix to use your version.

File: sunpy/util/config.py [hunk 1]
```diff
@@ -119,9 +119,13 @@ def _is_writable_dir(p):
     """
     Checks to see if a directory is writable.
     """
-    if not Path(p).exists():
-        Path(p).mkdir(parents=True)
-    return Path(p).is_dir() and os.access(p, os.W_OK)
+    # Worried about multiple threads creating the directory at the same time.
+    try:
+        Path(p).mkdir(parents=True, exist_ok=True)
+    except FileExistsError:
```
Comment (ayshih) [inline sunpy/util/config.py]:
```suggestion
    except FileExistsError:  # raised if there's an existing file instead of a directory
```

Comment (nabobalis) [reply sunpy/util/config.py]: I added this and expanded the test for it.

**Comment by nabobalis** [comment]
Thanks @ayshih

**Pull request merged**